testcases/BMCRedFish.py

```diff
@@ -0,0 +1,76 @@
+import time
+import subprocess
+import subprocess
+import re
+import sys
+
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+
+import unittest
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.OpTestEBMC import EBMCHostManagement
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class OpTestEBMCRedFish(unittest.TestCase):
+    '''
+    Reading or updating attributes from Redfish API
+    '''
+    @classmethod
+    def setUpClass(cls):
+        """
+        Pre setup before starting Redfish commands execution
+        """
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.cv_BMC = conf.bmc()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        if self.bmc_type == "EBMC_PHYP" :
+            self.is_lpar = True
+            self.hmc_user = conf.args.hmc_username
+            self.hmc_password = conf.args.hmc_password
+            self.hmc_ip = conf.args.hmc_ip
+            self.lpar_name = conf.args.lpar_name
+            self.system_name = conf.args.system_name
+            self.cv_HMC = self.cv_SYSTEM.hmc
+            self.bmc_ip = conf.args.bmc_ip
+            self.bmc_username = conf.args.bmc_username
+            self.bmc_password = conf.args.bmc_password
+            self.bios_attribute = conf.args.bios_attribute
+            self.bios_attribute_value = conf.args.bios_attribute_value
+            
+    def get_current_ioadapter_enlarged_capacity(self):
+        """
+        Get ioadapter enlarged capcity value
+        """
+        log.debug("=====Get current IOAdapter Enlarge Capacity=====")
+        return EBMCHostManagement.get_bios_attribute_value(
+            bios_attribute="hb_ioadapter_enlarged_capacity_current"
+        )
+            
+    def set_ioenlarge_capacity(self):
+        """
+        Set ioadapter enlarged capcity value
+        """
+        log.debug("=====Set IOAdapter Enlarge Capacity=====")
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        if self.get_current_ioadapter_enlarged_capacity() != self.bios_attribute_value:
+            if EBMCHostManagement.get_power_state() is "Off":
+                EBMCHostManagement.set_bios_attribute(
+                    bios_attribute=self.bios_attribute,
+                    bios_attribute_val=self.bios_attribute_value
+                )
+            else:
+                log.debug("Host power state is not Off," 
+                          f" Please power-off the host to re-try set attribute {self.bios_attribute}")
+        else:
+            log.debug("Provided bios attribute vaue is same a exisintg, " 
+                      f"ignoring set attribute for (self.bios_attribute)")
+    
```

Comment:
no need of sepearate py testcase... make sure you add the ioenlargecapacity code into MachineConfig.py  and remove BMCRedFish.py   

see if you can move get_current_ioadapter_enlarged_capacity() and  set_ioenlarge_capacity(self) to common utils or EBMC or MachineConfig.py